code/modules/food_and_drinks/food/snacks.dm

```diff
@@ -228,7 +165,62 @@
 			N.adjustBruteLoss(-1)
 			N.adjustFireLoss(-1)
 
+/obj/item/reagent_containers/food/snacks/sliceable/examine(mob/user)
+	..()
+	to_chat(user, "<span class='notice'>Alt-click to put something small inside.</span>")
+
+/obj/item/reagent_containers/food/snacks/sliceable/AltClick(mob/user)
+	var/obj/item/I = user.get_active_hand()
+	if(!I)
+		return
+	if(I.w_class > WEIGHT_CLASS_SMALL)
+		to_chat(user, "<span class='warning'>You cannot fit [I] in [src]!</span>")
+		return
+	var/newweight = GetTotalContentsWeight() + I.GetTotalContentsWeight() + I.w_class
+	if(newweight > MAX_WEIGHT_CLASS)
+		// Nope, no bluespace slice food
+		to_chat(user, "<span class='warning'>You cannot fit [I] in [src]!</span>")
+		return 1
+	if(!iscarbon(user))
+		return 1
+	to_chat(user, "<span class='warning'>You slip [I] inside [src].</span>")
+	user.unEquip(I)
```

Comment:
You need to check if the user can actually drop the item in their hand, otherwise `NODROP` items will cause bad things.
Do an `if(!user.drop_item())` check to do so.

Reply:
unEquip already accounts for it

``` 
	if(!canUnEquip(I, force))
		return 0
```

Reply:
throughout the condebase, what @Crazylemon64 is the typical check. Then to have a message of the thing being stuck to your hand if it fails.